domain/schema/controller.go

```diff
@@ -575,7 +575,21 @@ func controllerConfigSchema() schema.Patch {
 CREATE TABLE controller_config (
     key     TEXT PRIMARY KEY,
     value   TEXT
-);`)
+);
+
+CREATE TABLE controller (
+    uuid    TEXT PRIMARY KEY NOT NULL
```

Comment:
We should store name in here too.

Reply:
The name is optional according to the schema.

Reply:
For example: `juju controller-config controller-name=""`

Reply:
Controller name is now mandatory, and any code which allows it to be empty is a bug.
At bootstrap, we accept user input as arg[1] for the controller name if if that's missing, we default to
```
	if c.controllerName == "" {
		c.setControllerName(defaultControllerName(cloud.Name, region.Name))
	}
```

Reply:
Also, juju show-controller uses a map key which is the controller name, so it really is mandatory

Reply:
I wonder though, we currently support updating the controller name. So from that sense, it's config. But there's also been requests to rename models, and model name is considered a fundamental model entity attribute rather than config. I think controller name is in that category.

Reply:
> But there's also been requests to rename models, and model name is considered a fundamental model entity attribute rather than config. I think controller name is in that category.

Then we should come up with a design that fits the requirements for both model-config and controller-config.

Reply:
https://bugs.launchpad.net/juju/+bug/2065381